tests/unit/AnnualReport.spec.ts

```diff
@@ -2103,3 +2103,149 @@ describe('AnnualReport - Part 7 - Concurrent Saves', () => {
+            },
+            header: {
+              name: 'annualReport',
+              date: '2017-06-06',
+              submitter: 'cp0001191',
+              status: 'DRAFT',
+              certifiedBy: 'Full Name',
+              email: 'no_one@never.get',
+              filingId: 123
+            }
+          }
+        }
+      }
+    })
+
+    p1.catch(() => {}) // pre-empt "unhandled promise rejection" warning
+
+    sinon
+      .stub(axios, 'post')
+      .withArgs('businesses/CP0001191/filings')
+      .returns(p1)
+    
+    const localVue = createLocalVue()
+    localVue.use(VueRouter)
+    const router = mockRouter.mock()
+    router.push({ name: 'annual-report', params: { filingId: '0' } }) // new filing id
+
+    wrapper = mount(AnnualReport, {
+      store,
+      localVue,
+      router,
+      stubs: {
+        ArDate: true,
+        AgmDate: true,
+        OfficeAddresses: true,
+        Directors: true,
+        Certify: true,
+        StaffPayment: true,
+        Affix: true,
+        SbcFeeSummary: true,
+        ConfirmDialog: true,
+        PaymentErrorDialog: true,
+        ResumeErrorDialog: true,
+        SaveErrorDialog: true,
+        BcolErrorDialog: true
+      },
+      vuetify
+    })
+    vm = wrapper.vm
+  })
+
+  afterEach(() => {
+    sinon.restore()
+    wrapper.destroy()
+  })
+
+  it('Attempt to file and pay with a BCOL error', async () => {
+    // set necessary session variables
+    sessionStorage.setItem('BASE_URL', `${process.env.VUE_APP_PATH}/`)
+    sessionStorage.setItem('PAY_API_URL', '')
+    sessionStorage.setItem('AUTH_URL', 'auth/')
+    const get = sinon.stub(axios, 'get')
+
+    get.withArgs('businesses/CP0001191/tasks')
+    .returns(new Promise(resolve => resolve({ data: { tasks: [] } })))
+
+    // make sure form is validated
+    vm.staffPaymentFormValid = true
+    vm.agmDateValid = true
+    vm.addressesFormValid = true
+    vm.directorFormValid = true
+    vm.certifyFormValid = true
+    vm.directorEditInProgress = false
+    store.state.filingData = [{ filingTypeCode: 'OTANN', entityType: 'CP' }] // dummy data
+
+    // stub address data
+    vm.addresses = {
+      registeredOffice: {
+        deliveryAddress: {},
+        mailingAddress: {}
+      }
+    }
+
+    // make sure a fee is required
+    vm.totalFee = 100
+
+    // sanity check
+    expect(vm.bcolObj).toBeNull()
+
+    const button = wrapper.find('#ar-file-pay-btn')
+    expect(button.attributes('disabled')).toBeUndefined()
+    
+    get.withArgs('codes/errors/BCOL_ERROR')
+    .returns(new Promise(resolve => resolve({ data: {
+      detail: 'An Error has occured',
+      title: 'Error'
+    }})))
+    // click the File & Pay button
+    await button.trigger('click')
+    await flushPromises()
+    await vm.onClickFilePay()
+    // work-around because click trigger isn't working
+
+    // verify an error has been received
+    expect(vm.bcolObj?.detail?.length).toBeGreaterThan(0)
```

Comment:
Please create a new unit test file for your new dialog. The tests should verify that the different props generate the expected function/output.

Reply:
I considered this, but I didn't see a test file for paymenterrordialog. Will do

Reply:
There are other files for other dialogs. Check them out for guidance.

Reply:
Done, great!